# Aman Rohada

Business Analytics(2022-2024) | 021230022987 | Introduction to Machine Learning | Analysis(Test) completed - 3(One-Sample t-test, Two-Sample t-test and Pearson Correlation )




In [1]:
import pandas as pd
import numpy as np
from scipy.stats  import ttest_1samp,ttest_ind,ttest_rel
import researchpy as rp
from scipy import stats

In [2]:
data=pd.read_csv(r"C:\Users\amanr\AppData\Local\Temp\Temp1_archive (4).zip\Attrition Data.csv")
data

Age Attrition              Department  DistanceFromHome  Education  \
0      41       Yes                   Sales                 1          2   
1      49        No  Research & Development                 8          1   
2      37       Yes  Research & Development                 2          2   
3      33        No  Research & Development                 3          4   
4      27        No  Research & Development                 2          1   
...   ...       ...                     ...               ...        ...   
1465   36        No  Research & Development                23          2   
1466   39        No  Research & Development                 6          1   
1467   27        No  Research & Development                 4          3   
1468   49        No                   Sales                 2          3   
1469   34        No  Research & Development                 8          3   

     EducationField  EnvironmentSatisfaction  JobSatisfaction MaritalStatus  \
0     Life Sciences                        2                4        Single   
1     Life Sciences                        3                2       Married   
2             Other                        4                3        Single   
3     Life Sciences                        4                3       Married   
4           Medical                        1                2       Married   
...             ...                      ...              ...           ...   
1465        Medical                        3                4       Married   
1466        Medical                        4                1       Married   
1467  Life Sciences                        2                2       Married   
1468        Medical                        4                2       Married   
1469        Medical                        2                3       Married   

      MonthlyIncome  NumCompaniesWorked  WorkLifeBalance  YearsAtCompany  
0              5993                   8                1               6  
1              5130                   1                3              10  
2              2090                   6                3               0  
3              2909                   1                3               8  
4              3468                   9                3               2  
...             ...                 ...              ...             ...  
1465           2571                   4                3               5  
1466           9991                   4                3               7  
1467           6142                   1                3               6  
1468           5390                   2                2               9  
1469           4404                   2                4               4  

[1470 rows x 13 columns]

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Age                      1470 non-null   int64 
 1   Attrition                1470 non-null   object
 2   Department               1470 non-null   object
 3   DistanceFromHome         1470 non-null   int64 
 4   Education                1470 non-null   int64 
 5   EducationField           1470 non-null   object
 6   EnvironmentSatisfaction  1470 non-null   int64 
 7   JobSatisfaction          1470 non-null   int64 
 8   MaritalStatus            1470 non-null   object
 9   MonthlyIncome            1470 non-null   int64 
 10  NumCompaniesWorked       1470 non-null   int64 
 11  WorkLifeBalance          1470 non-null   int64 
 12  YearsAtCompany           1470 non-null   int64 
dtypes: int64(9), object(4)
memory usage: 149.4+ KB


In [4]:
data.isnull().sum()

Age                        0
Attrition                  0
Department                 0
DistanceFromHome           0
Education                  0
EducationField             0
EnvironmentSatisfaction    0
JobSatisfaction            0
MaritalStatus              0
MonthlyIncome              0
NumCompaniesWorked         0
WorkLifeBalance            0
YearsAtCompany             0
dtype: int64

In [5]:
data.isnull().sum().sum()

0

In [6]:
data = data.drop(columns=['Education','EducationField','EnvironmentSatisfaction','NumCompaniesWorked'])

In [7]:
data

Age Attrition              Department  DistanceFromHome  \
0      41       Yes                   Sales                 1   
1      49        No  Research & Development                 8   
2      37       Yes  Research & Development                 2   
3      33        No  Research & Development                 3   
4      27        No  Research & Development                 2   
...   ...       ...                     ...               ...   
1465   36        No  Research & Development                23   
1466   39        No  Research & Development                 6   
1467   27        No  Research & Development                 4   
1468   49        No                   Sales                 2   
1469   34        No  Research & Development                 8   

      JobSatisfaction MaritalStatus  MonthlyIncome  WorkLifeBalance  \
0                   4        Single           5993                1   
1                   2       Married           5130                3   
2                   3        Single           2090                3   
3                   3       Married           2909                3   
4                   2       Married           3468                3   
...               ...           ...            ...              ...   
1465                4       Married           2571                3   
1466                1       Married           9991                3   
1467                2       Married           6142                3   
1468                2       Married           5390                2   
1469                3       Married           4404                4   

      YearsAtCompany  
0                  6  
1                 10  
2                  0  
3                  8  
4                  2  
...              ...  
1465               5  
1466               7  
1467               6  
1468               9  
1469               4  

[1470 rows x 9 columns]

# 1.ONE SAMPLE T-TEST

# Hypothesis

* H0 : There is no significant difference between the sample mean and population mean with regards to Attrition and Years At            Company. 
* H1 : There is significant difference between the sample mean and population mean with regards to Attrition and Years At              Company. 
 

In [8]:
data.Attrition.value_counts()

No     1233
Yes     237
Name: Attrition, dtype: int64

In [9]:
attrition_group = data[data['Attrition']=='Yes']
attrition_group

Age Attrition              Department  DistanceFromHome  \
0      41       Yes                   Sales                 1   
2      37       Yes  Research & Development                 2   
14     28       Yes  Research & Development                24   
21     36       Yes                   Sales                 9   
24     34       Yes  Research & Development                 6   
...   ...       ...                     ...               ...   
1438   23       Yes                   Sales                 9   
1442   29       Yes  Research & Development                 1   
1444   56       Yes  Research & Development                 7   
1452   50       Yes                   Sales                 1   
1461   50       Yes                   Sales                28   

      JobSatisfaction MaritalStatus  MonthlyIncome  WorkLifeBalance  \
0                   4        Single           5993                1   
2                   3        Single           2090                3   
14                  3        Single           2028                3   
21                  1        Single           3407                3   
24                  1        Single           2960                3   
...               ...           ...            ...              ...   
1438                1       Married           1790                2   
1442                4       Married           4787                4   
1444                3       Married           2339                1   
1452                3      Divorced           6728                3   
1461                1      Divorced          10854                3   

      YearsAtCompany  
0                  6  
2                  0  
14                 4  
21                 5  
24                 4  
...              ...  
1438               1  
1442               2  
1444              10  
1452               6  
1461               3  

[237 rows x 9 columns]

In [10]:
mean_attrition = attrition_group['YearsAtCompany'].mean()
mean_attrition

5.1308016877637135

In [11]:
mean_population = data['YearsAtCompany'].mean()

t_value,p_value = ttest_1samp(attrition_group['YearsAtCompany'], mean_population)
t_value,p_value
print('p-value is:',p_value*1000)

p-value is: 0.0021668987768796405


# Conclusion
* Since p-value is  0.0021668987768796405,which is less than 0.05 we reject H0.
* Therefore we can colcude that,There is significant difference between the sample mean and population mean with regards to Attrition and Years At Company. 

# 2.TWO SAMPLE T-TEST

# Hypothesis

* H0 : There is no difference in job satisfaction between employees with a high work-life balance(WorkLifeBalance=4) and                employees with a low work-life balance(WorkLifeBalance=1)
* H1 : There is difference in job satisfaction between employees with a high work-life balance(WorkLifeBalance=4) and employees        with a low work-life balance(WorkLifeBalance=1)

In [12]:
wor_bal_1 = data[data['WorkLifeBalance']== 1]
wor_bal_1

Age Attrition              Department  DistanceFromHome  \
0      41       Yes                   Sales                 1   
91     51        No                   Sales                21   
95     54        No  Research & Development                 2   
139    30        No         Human Resources                 9   
152    53        No                   Sales                 6   
...   ...       ...                     ...               ...   
1355   36        No                   Sales                17   
1380   35        No                   Sales                18   
1421   47        No  Research & Development                 1   
1444   56       Yes  Research & Development                 7   
1460   29        No  Research & Development                28   

      JobSatisfaction MaritalStatus  MonthlyIncome  WorkLifeBalance  \
0                   4        Single           5993                1   
91                  4        Single           5441                1   
95                  3       Married          13549                1   
139                 4       Married           6347                1   
152                 3       Married           2306                1   
...               ...           ...            ...              ...   
1355                2       Married           5507                1   
1380                1       Married           5561                1   
1421                2       Married          11957                1   
1444                3       Married           2339                1   
1460                1        Single           3785                1   

      YearsAtCompany  
0                  6  
91                10  
95                 4  
139               11  
152                7  
...              ...  
1355               4  
1380               5  
1421              13  
1444              10  
1460               5  

[80 rows x 9 columns]

In [13]:
wor_bal_4 = data[data['WorkLifeBalance']== 4]
wor_bal_4

Age Attrition              Department  DistanceFromHome  \
31     44        No  Research & Development                10   
33     39       Yes                   Sales                 5   
38     36        No  Research & Development                 5   
59     37        No  Research & Development                 1   
77     45        No  Research & Development                 6   
...   ...       ...                     ...               ...   
1409   40        No  Research & Development                11   
1423   22        No  Research & Development                 1   
1442   29       Yes  Research & Development                 1   
1456   35        No  Research & Development                18   
1469   34        No  Research & Development                 8   

      JobSatisfaction MaritalStatus  MonthlyIncome  WorkLifeBalance  \
31                  4       Married           6465                4   
33                  4       Married           2086                4   
38                  1       Married           3419                4   
59                  3      Divorced           5993                4   
77                  1       Married          13245                4   
...               ...           ...            ...              ...   
1409                3       Married           6323                4   
1423                3        Single           3375                4   
1442                4       Married           4787                4   
1456                3       Married           5689                4   
1469                3       Married           4404                4   

      YearsAtCompany  
31                 4  
33                 1  
38                 1  
59                 7  
77                 0  
...              ...  
1409              10  
1423               3  
1442               2  
1456              10  
1469               4  

[153 rows x 9 columns]

In [14]:
mean_jobsat_wlbalance1 = data[data['WorkLifeBalance'] == 1]['JobSatisfaction'].mean()
mean_jobsat_wlbalance1

2.65

In [15]:
mean_jobsat_wlbalance4 = data[data['WorkLifeBalance'] == 4]['JobSatisfaction'].mean()
mean_jobsat_wlbalance4

2.7450980392156863

In [16]:
jobsat_wlbalance1 = data[data['WorkLifeBalance'] == 1]['JobSatisfaction']
jobsat_wlbalance4 = data[data['WorkLifeBalance'] == 4]['JobSatisfaction']
t_stat, p_value = ttest_ind(jobsat_wlbalance1, jobsat_wlbalance4)
print("t-statistic:", t_stat)
print("p-value:", p_value)

t-statistic: -0.6224415525791399
p-value: 0.5342654955372359


# Conclusion

* Since p-value = 0.535, which is gretaer than the level of significance(0.05) we fail to reject H0.
* Hence we conclude there is no difference in job satisfaction between employees with a high work-life           balance(WorkLifeBalance=4) and employees with a low work-life balance(WorkLifeBalance=1) 

# 3.PEARSON'S CORRELATION

# ASSUMPTIONS

* Observations in each sample are independent and identically distributed (iid).
* Observations in each sample are normally distributed. 
* Observations in each sample have the same variance.

# Hypothesis

* H0 : There is no significant correlation between Age and Job Satisfaction.
* H1 : There is significant correlation between Age and Job Satisfaction.


In [17]:
from scipy.stats import pearsonr
age = data['Age']
JobSat = data['JobSatisfaction']
# TO calculate the correlation coefficient and p-value
corr, p_value = pearsonr(age, JobSat)
print("Correlation coefficient:", corr)
print("P-value:", p_value)

Correlation coefficient: -0.004891877148687333
P-value: 0.8513478273017392


# Conclusion

* As Correlation coefficient: -0.0049 which is approximately equal to 0, so we can say that there is no correlation between age   and Job satisfaction
* Since p-value = 0.851 is gretaer than level of significance(0.05) we fail to reject H0.
* Therefore we can conclude that, there is no significant correlation between Age and Job Satisfaction.